baybe/recommenders/pure/bayesian/base.py

```diff
@@ -47,75 +54,53 @@ def _validate_deprecated_argument(self, _, value) -> None:
-            train_x: The features of the conducted experiments.
-            train_y: The corresponding response values.
-
-        Raises:
-            NotImplementedError: If the setup is attempted from empty training data
         """
-        if train_x is None or train_y is None:
-            raise NotImplementedError(
-                "Bayesian recommenders do not support empty training data yet."
-            )
-
-        best_f = train_y.max().item()
-        surrogate_model = self._fit(searchspace, train_x, train_y)
+        best_f = self._train_y.max().item()
+        self.surrogate_model.fit(searchspace, *to_tensor(self._train_x, self._train_y))
         self._botorch_acqf = self.acquisition_function.to_botorch(
-            surrogate_model, best_f
+            self.surrogate_model, best_f
         )
 
-    def _fit(
-        self,
-        searchspace: SearchSpace,
-        train_x: pd.DataFrame,
-        train_y: pd.DataFrame,
-    ) -> Surrogate:
-        """Train a fresh surrogate model instance.
-
-        Args:
-            searchspace: The search space.
-            train_x: The features of the conducted experiments.
-            train_y: The corresponding response values.
-
-        Returns:
-            A surrogate model fitted to the provided data.
-
-        Raises:
-            ValueError: If the training inputs and targets do not have the same index.
-        """
-        # validate input
-        if not train_x.index.equals(train_y.index):
-            raise ValueError("Training inputs and targets must have the same index.")
-
-        self.surrogate_model.fit(searchspace, *to_tensor(train_x, train_y))
-
-        return self.surrogate_model
-
     def recommend(  # noqa: D102
         self,
         searchspace: SearchSpace,
+        objective: Optional[Objective],
         batch_size: int = 1,
-        train_x: Optional[pd.DataFrame] = None,
-        train_y: Optional[pd.DataFrame] = None,
+        train_df: Optional[pd.DataFrame] = None,
     ) -> pd.DataFrame:
         # See base class.
 
+        if objective is None:
+            raise NotImplementedError(
+                f"Recommenders of type '{BayesianRecommender.__name__}' require "
+                f"that an objective is specified."
+            )
+
+        if train_df is None:
+            raise NotImplementedError(
+                f"Recommenders of type '{BayesianRecommender.__name__}' do not support "
+                f"empty training data yet."
+            )
+
         if _ONNX_INSTALLED and isinstance(self.surrogate_model, CustomONNXSurrogate):
             CustomONNXSurrogate.validate_compatibility(searchspace)
 
-        self._setup_botorch_acqf(searchspace, train_x, train_y)
+        self._transform_data(train_df, searchspace, objective)
+        self._setup_botorch_acqf(searchspace)
 
-        return super().recommend(searchspace, batch_size, train_x, train_y)
+        return super().recommend(searchspace, objective, batch_size, train_df)
```

Comment:
since recommend  got a more complex signature its imo better to work with keywords here and not rely on a more or less arb order of args

Reply:
your wish is my command (note, though, that mypy kept us safe anyway ...)